src/services/pegout-data.processor.ts

```diff
@@ -49,7 +67,85 @@ export class PegoutDataProcessor implements FilteredBridgeTransactionProcessor {
     return events.some(event => event.name === BRIDGE_EVENTS.RELEASE_REQUEST_REJECTED);
   }
 
-  private processReleaseRequestReceivedStatus(extendedBridgeTx: ExtendedBridgeTx): void {
+  hasReleaseRequestedEvent(events: BridgeEvent[]): boolean {
+    return events.some(event => event.name === BRIDGE_EVENTS.RELEASE_REQUESTED);
+  }
+
+  private async processReleaseRequestedStatus(extendedBridgeTx: ExtendedBridgeTx): Promise<void> {
+    const events: BridgeEvent[] = extendedBridgeTx.events;
+    const releaseRequestedEvent = events.find(event => event.name === BRIDGE_EVENTS.RELEASE_REQUESTED);
+    if(!releaseRequestedEvent) {
+      return;
+    }
+
+    const originatingRskTxHash = <string> releaseRequestedEvent.arguments.get('rskTxHash');
+    const btcTxHash = <string> releaseRequestedEvent.arguments.get('btcTxHash');
+
+    const foundPegoutStatus = await this.pegoutStatusDataService.getLastByOriginatingRskTxHash(originatingRskTxHash);
+
+    if(!foundPegoutStatus) {
+      return this.logger.warn(`[processReleaseRequestedStatus] could not find a pegout status record
+       in the db for this transaction '${extendedBridgeTx.txHash}' with 'release_requested' event.`);
+    }
+
+    const newPegoutStatus: PegoutStatusDataModel = new PegoutStatusDataModel();
+
+    newPegoutStatus.btcRecipientAddress = foundPegoutStatus.btcRecipientAddress; 
+    newPegoutStatus.originatingRskTxHash = originatingRskTxHash;
+    newPegoutStatus.valueRequestedInSatoshis = foundPegoutStatus.valueRequestedInSatoshis;
+    newPegoutStatus.rskSenderAddress = foundPegoutStatus.rskSenderAddress;
+    newPegoutStatus.rskTxHash = extendedBridgeTx.txHash;
+    newPegoutStatus.rskBlockHeight = extendedBridgeTx.blockNumber;
+    newPegoutStatus.createdOn = extendedBridgeTx.createdOn;
+    newPegoutStatus.btcTxHash = btcTxHash;
+    newPegoutStatus.status = PegoutStatus.WAITING_FOR_CONFIRMATION;
+
+    await this.addValueInSatoshisToBeReceivedAndFee(newPegoutStatus);
+
+    try {
+      await this.pegoutStatusDataService.set(newPegoutStatus);
+      this.logger.trace(`[processReleaseRequestedStatus] pegout status for ${originatingRskTxHash} updated.`);
+    } catch(e) {
+      this.logger.warn('[processReleaseRequestedStatus] There was a problem with the storage', e);
+    }
+
+  }
+
+  private async addValueInSatoshisToBeReceivedAndFee(pegoutStatus: PegoutStatusDataModel): Promise<void> {
+    const rskTxHash = remove0x(pegoutStatus.originatingRskTxHash);
+    const bridgeState = await this.bridgeService.getBridgeState();
```

Comment:
it may make sense to run this command just once at the `process` method level

Reply:
Some of the statuses will not need this call. If we do it by default, then we would be waiting processing.